pac4j-core/src/main/java/org/pac4j/core/client/IndirectClientV2.java

```diff
@@ -49,7 +50,18 @@ protected C retrieveCredentials(final WebContext context) throws HttpAction {
             this.authenticator.validate(credentials, context);
             return credentials;
         } catch (CredentialsException e) {
-            logger.error("Failed to retrieve or validate credentials", e);
+            if (e instanceof MultipleAccountsFoundException) {
+                // MultipleAccountsFoundException is a subclass of CredentialsException even though
+                // it is not caused by an authentication error but by a a bug, so we consider it an error
+                throw e;
+            }
+            
+            if (logger.isDebugEnabled()) {
```

Comment:
This entire block appears to be duplicated here. Is there a way we could remove that duplication?